### PR TITLE
Upgrade to babel 7

### PR DESCRIPTION
**Pull request by johanhenrikssn**
**Summary**

Update assisted by `babel-upgrade` tool. This also moves away from using the [deprecated stage-0 preset](https://babeljs.io/blog/2018/07/27/removing-babels-stage-presets) and instead explicitly add each propsal as a dependency.

**Test plan**

No new tests added.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,16 +53,32 @@
...
+    "@babel/plugin-proposal-optional-chaining": "^7.0.0",
+    "@babel/plugin-proposal-pipeline-operator": "^7.0.0",
+    "@babel/plugin-proposal-throw-expressions": "^7.0.0",
+    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
+    "@babel/plugin-syntax-import-meta": "^7.0.0",
+    "@babel/plugin-transform-runtime": "^7.0.0",
+    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-flow": "^7.0.0",
+    "babel-core": "^7.0.0-bridge.0",
     "babel-eslint": "^7.2.3",
```
Comment (milesj) [inline package.json]: This should probably be `^9.0.0`.

File: package.json [hunk 2]
```diff
@@ -53,16 +53,32 @@
...
-    "babel-loader": "^6.2.5",
+    "babel-loader": "^8.0.0",
     "babel-plugin-array-includes": "^2.0.3",
     "babel-plugin-transform-builtin-extend": "^1.1.2",
     "babel-plugin-transform-inline-imports-commonjs": "^1.0.0",
```
Comment (milesj) [inline package.json]: This isn't compatible with v7 and can be replaced with `['@babel/plugin-transform-modules-commonjs', {allowTopLevelThis: true}],`.

File: package.json [hunk 3]
```diff
@@ -53,16 +53,32 @@
...
-    "babel-plugin-transform-runtime": "^6.4.3",
-    "babel-preset-env": "^1.6.0",
-    "babel-preset-flow": "^6.23.0",
-    "babel-preset-stage-0": "^6.0.0",
     "babylon": "^6.5.0",
```
Comment (milesj) [inline package.json]: This is now `@babel/parser`.

File: package.json [hunk 4]
```diff
@@ -53,16 +53,32 @@
...
+    "@babel/plugin-proposal-numeric-separator": "^7.0.0",
+    "@babel/plugin-proposal-optional-chaining": "^7.0.0",
+    "@babel/plugin-proposal-pipeline-operator": "^7.0.0",
+    "@babel/plugin-proposal-throw-expressions": "^7.0.0",
+    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
+    "@babel/plugin-syntax-import-meta": "^7.0.0",
+    "@babel/plugin-transform-runtime": "^7.0.0",
+    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-flow": "^7.0.0",
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (milesj) [inline package.json]: It's best to also place this in Yarn `resolutions`.

Comment (arcanis) [reply package.json]: Why is the bridge needed?

**Comment by arcanis** [comment]
Thanks for taking this! Can you rebase on master? I also noticed that some tests are failing - seems like modules are not properly transpiled

File: package.json (hunk 4 above)
Comment (milesj) [reply package.json]: Because jest requires `babel-core` and not `@babel/core`, so it needs to be bridged.

**Pull request closed without merging**